File: NetBeans/src/edu/hdsb/gwss/nathan/ics3u/u7/Golf.java
/*
Name:Nathan Wu
Date: 5/23/28
Version: 0.01
Description:
    This program takes your inputed golf scores and calculates the total, max 
    and min score.
 */
package edu.hdsb.gwss.nathan.ics3u.u7;

import java.util.Scanner;
import java.util.StringTokenizer;

/**
 *
 * @author nwu10
 */
public class Golf {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        int[] hole = new int[18];
        splashScreen();
        StringTokenizer str = new StringTokenizer(data(hole));
        System.out.println("Total: " + str.nextToken());
        System.out.println("Max: " + str.nextToken());
        System.out.println("Min: " + str.nextToken());
    }

    public static void splashScreen(){
        System.out.println("Golf");
        System.out.println("---------");
        System.out.println("This program takes your inputed golf scores and "
                + "calculates the total, max and min score.");
    }
    public static String data(int data[]) {
        //objects
        Scanner input = new Scanner(System.in);

        int max = 0, min = 0, total = 0;
        for (int i = 0; i < data.length; i++) {
            //user input
            System.out.println("Hole " + (i + 1) + ": ");
            data[i] = input.nextInt();

            //user input
            max = Math.max(data[i], max);
            if (i == 0) {
                min = Math.min(data[i], max);
            } else {
                min = Math.min(data[i], min);
            }
            total = total + data[i];
        }
        return total + " " + max + " " + min;
    }

}

File: independant_study/Golf.py
def dataMethod(data):
    # variables
    max, total = 0, 0, 0
    for x in data:
        i = data.index(x)
        # get input
        data[i] = int(input("Hole " + str(i + 1) + ": "))
        # determine if it's max
        if data[i] > max:
            max = data[i]
        # determine if it's min
        if i == 0:
            if data[i] < max:
                min = data[i]
            else:
                min = max
        else:
            if data[i] < min:
                min = data[i]
        total = total + data[i]
    # return
    return [total, max, min]


# How many holes
hole = [0] * 5
# get scores and find results
score = dataMethod(hole)
# print results
print("Total: " + str(score[0]))
print("Max: " + str(score[1]))
print("Min: " + str(score[2]))
